### DISCUSSION TITLE
[c64dtv] Exclude memory from em driver, possible?

### DISCUSSION THREAD

**Discussion by megatron-uk** [cc65/cc65#1979] [Q&A]
I am playing around with my old DTV2 system and looking at developing some code to take advantage of the extra memory and linear framebuffer features - I'm not sure why I didn't try building anything for the DTV in the past; it's got some really cool extensions to the C64 (although some of them being bugged).

I have a few use cases where I want to access the (64kb-2048KB) extended memory region directly using the DMA engine and/or blitter for moving content around a double-buffered framebuffer... however, I'd also like to take advantage of the DTV extended memory driver for loading code overlays into the rest of the high memory area etc; that's something I _don't_ want to have to invent myself!

Is there any mechanism to inform the extended memory driver to exclude one or more memory regions; a simple example being to instruct the driver to allocate pages only from above 0x030000 (the 64kb base, plus 2x 64kb framebuffer regions). Or, is there a simple modification I can make to the config passed to ld65 to exclude a given region?

I'm not an assembly programmer, so I really don't want to have to go editing the highmem em driver.

**Comment by kugelfuhr** [comment]
The DTV himem driver uses memory starting at $20000. With a total of 2MB RAM minus 128KB reserved space, it returns 7680 as the number of available pages. Each page is 256 bytes. Page zero starts at $20000, page 1 at $20100 and so on.

So parallel usage is easy. All you have to do is not to pass page numbers that refer to memory your program uses for other purposes. This is true for most other drivers as well.

**Comment by megatron-uk** [comment]
Ah, that's great; so as long as I skip 'N' pages I'll be fine to mix use of the high memory in this way?

**Comment by kugelfuhr** [comment]
From a programmers standpoint, it might be easier to just limit the maximum page count and use the upper pages for something else. But regardless of how you do it: Any page not accessed via the driver can be used for other purposes.
